src/core/utils/xmlParse.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @prettier
+ */
+
+const traverseXML = (node, result = {}) => {
+  if (node.nodeType === 1) {
+    // ELEMENT_NODE - parameter
+    const name = node.nodeName
+
+    if (node.nodeName === "parsererror") {
+      throw new Error("Parameter string value must be valid XML")
+    }
+
+    const childNodes = Array.from(node.childNodes)
```

Comment:
We might need to get rid of nodes with only whitespace characters here, as at this point our result is this: 
```
{
    "0": "\n",
    "1": "\t",
    "id": "10",
    "name": "doggie",
    "category": {
        "0": "\n",
        "1": "\t",
        "2": "\t",
        "id": "1",
        "name": "Dogs"
    }
}
```
The validation for this will pass but perhaps it might create issues later on.